src/pages/dogs/[id].jsx

```diff
@@ -13,175 +19,318 @@ import Image from "next/image";
+                  <div className="flex gap-4">
+                    <Chip
+                      label={dog.location}
+                      type={ChipTypeStyles[dog.location] || ChipTypeStyles.Tag}
                     />
+                    <div className="flex justify-center items-center space-x-2">
+                      <Image
+                        priority
+                        src={dog.gender === "Male" ? maleicon : femaleicon}
+                        alt="Male Dog"
+                      />
 
-                    <div>Male</div>
+                      <div>Male</div>
+                    </div>
                   </div>
                 </div>
-              </div>
 
-              <div className="pt-6 pl-1 font-bold text-3xl">{dog.name}</div>
-              <div className="flex space-x-16">
-                <div className="flex-col pt-8 pl-1 text-lg space-y-2">
-                  <div>
-                    Birth Date:{" "}
-                    {dateutils.getDateString(new Date(dog.dateOfBirth))}
-                  </div>
-                  <div>Sex: {dog.gender}</div>
+                {!isEdit && (
+                  <div className="pt-6 pl-1 font-bold text-3xl">{dog.name}</div>
+                )}
 
-                  <div>Breed: {dog.breed}</div>
-                  <div>Coat Color: N/A</div>
-                </div>
+                <div className="flex space-x-16">
+                  <div className="flex-col pt-8 pl-1 text-lg space-y-2">
+                    {isEdit && (
+                      <FormField
+                        className="h-min pl-1 font-bold text-3xl"
+                        label="Name"
+                        keyLabel={"name"}
+                        errors={errors}
+                        isEditing={isEdit}
+                        value={dog.name}
+                        register={register}
+                        showLabel={false}
+                      />
+                    )}
+
+                    <FormField
+                      isEditing={isEdit}
+                      label={"Birth Date"}
+                      errors={errors}
+                      keyLabel={"dateOfBirth"}
+                      register={register}
+                    >
+                      {dateutils.getDateString(new Date(dog.dateOfBirth))}
+                    </FormField>
+                    <FormField
+                      isEditing={isEdit}
+                      register={register}
+                      label={"Sex"}
+                      errors={errors}
+                      keyLabel={"gender"}
+                      value={dog.gender}
+                    />
+                    <FormField
+                      isEditing={isEdit}
+                      register={register}
+                      label={"Breed"}
+                      errors={errors}
+                      keyLabel={"breed"}
+                      value={dog.breed}
+                    />
+                    <FormField
```

Comment:
coat color and inputs on the information page are not clickable in deploy preview. Ik I said to leave the complex ones uneditable, but any that are strings or numbers like `litterSize`, `litterComposition`, feeding section, etc can be implemented now

Reply:
will update tn!

Reply:
so objects and numbers are also weird right now because its been kind of difficult to validate the type of data and zod. for example, since we are using register function from react-hook-form, numbers are treated as strings in the input and this doesn't parse well with fields that are zod.number(). so i have to figure out how to do these as well later.

right now, inputs that are only purely strings and enums in the database (not objects, numbers, or arrays) will be able to be modified

Reply:
I will for sure fix this up when I deal with more complex inputs and see how to make it work with zod better